### DISCUSSION TITLE
Allow `dagger <func name>` vs `dagger call <func name>` if no conflict

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#12315] [Ideas]
I often find myself wanting to run
```
dagger build
```
or similar and then I have to go back and add the `call`. Ideally, this would just work if there was no confusion.

cc @shykes

**Comment by sipsma** [comment]
I think this would align with one of the possibilities being discussed in this issue: https://github.com/dagger/dagger/issues/6947#issuecomment-2135899729

I'd say that I'm cautiously intrigued by this possibility; there's trickiness around conflicts and such but if we can get it to work it would be a nice improvement IMO.

**Comment by shykes** [comment]
That's a bridge too far for me, because it creates too much variability in the root commands, from one config to the next.

But there are two avenues we can pursue:

1. Override `dagger run FOO` so that, if `FOO` is either an installed module, or a function in the current module, that gets called instead of a local command. If `FOO` is both a high-level function and a dependency, an error is returned. The default behavior of `dagger run` would remain the same, as a fallback. Note, this is similar to `npm run FOO`, so there's a familiarity element to it.

2. Scripting. I still want `dagger` to be a shell, with a bash-compatible syntax for starters, but opening to adding more languages later, too. In that scripting syntax, you can just run `build`.

**Comment by samalba** [comment]
I would keep as-is (`dagger call FUNC`). Sometimes I miss the `call` command and it fails, but it's better to keep it explicit. The verb `call` makes sense for a function. Giving an option to omit the verb will only introduce ambiguity.
